### PR TITLE
Wrap experimental commands in -H:+-UnlockExperimentalVMOption where necessary

### LINKED ISSUE #187: [23.1] nativeJVMTextProcessing integration test fails with experimental warning
It fails with:

```
Tests run: 12, Failures: 3, Errors: 0, Skipped: 5, Time elapsed: 221.591 s <<< FAILURE! - in org.graalvm.tests.integration.AppReproducersTest
nativeJVMTextProcessing{TestInfo}  Time elapsed: 42.007 s  <<< FAILURE!
org.opentest4j.AssertionFailedError:
build-and-run.log log should not contain error or warning lines that are not whitelisted. See /disk/graal/upstream-sources/mandrel-integration-tests/testsuite/target/archived-logs/org.graalvm.tests.integration.AppReproducersTest/nativeJVMTextProcessing/build-and-run.log and check these offending lines:
Warning: The option '-H:+PreserveFramePointer' is experimental and must be enabled via '-H:+UnlockExperimentalVMOptions' in the future.
Warning: The option '-H:-DeleteLocalSymbols' is experimental and must be enabled via '-H:+UnlockExperimentalVMOptions' in the future.
Warning: The option '-H:+TrackNodeSourcePosition' is experimental and must be enabled via '-H:+UnlockExperimentalVMOptions' in the future.
Warning: The option '-H:GenerateDebugInfo=1' is experimental and must be enabled via '-H:+UnlockExperimentalVMOptions' in the future.
Warning: The option '-H:+OmitInlinedMethodDebugLineInfo' is experimental and must be enabled via '-H:+UnlockExperimentalVMOptions' in the future.
Warning: The option '-H:+DebugCodeInfoUseSourceMappings' is experimental and must be enabled via '-H:+UnlockExperimentalVMOptions' in the future.
Warning: Please re-evaluate whether any experimental option is required, and either remove or unlock it. The build output lists all active experimental options, including where they come from and possible alternatives. If you think an experimental option should be considered as stable, please file an issue. ==> expected: <true> but was: <false>
        at org.graalvm.tests.integration.AppReproducersTest.nativeJVMTextProcessing(AppReproducersTest.java:909)
```

### PR DESCRIPTION
**Pull request by zakkak**
Related to #184 

Closes #187 

Fixes part of #188 and #190

### CODE REVIEW

File: testsuite/src/it/java/org/graalvm/tests/integration/utils/BuildAndRunCmds.java [hunk 1]
```diff
@@ -206,10 +206,12 @@ public enum BuildAndRunCmds {
                     :
                     new String[]{"unzip", "test_data.txt.zip", "-d", "target"},
 
-            new String[]{"native-image", "-H:GenerateDebugInfo=1", "-H:+PreserveFramePointer", "-H:-DeleteLocalSymbols",
+            new String[]{"native-image", DebugSymbolsTest.DebugOptions.UnlockExperimentalVMOptions_23_1.token,
+                    "-H:GenerateDebugInfo=1", "-H:+PreserveFramePointer", "-H:-DeleteLocalSymbols",
```
[the remark this answers is not in the document]
Comment (zakkak) [reply testsuite/src/it/java/org/graalvm/tests/integration/utils/BuildAndRunCmds.java:210]: I think we have discussed this in the past and @Karm wanted to have fine control of the options.

@Karm WDYT?

File: testsuite/src/it/java/org/graalvm/tests/integration/DebugSymbolsTest.java [hunk 2]
```diff
@@ -212,12 +222,18 @@ public void debugSymbolsQuarkus(TestInfo testInfo) throws IOException, Interrupt
...
                 runCommand(getRunCommand("git", "apply", "quarkus_3.x.patch"),
-                        Path.of(BASE_DIR, Apps.QUARKUS_FULL_MICROPROFILE.dir).toFile());
+                        Path.of(BASE_DIR, app.dir).toFile());
             }
 
             // Build
             processLog = Path.of(appDir.getAbsolutePath(), "logs", "build-and-run.log").toFile();
-            builderRoutine(app.buildAndRunCmds.cmds.length - 1, app, report, cn, mn, appDir, processLog);
+            final Map<String, String> switches;
+            if (UsedVersion.getVersion(false).compareTo(Version.create(23, 1, 0)) >= 0) {
```
[the remark this answers is not in the document]
Comment (zakkak) [reply testsuite/src/it/java/org/graalvm/tests/integration/DebugSymbolsTest.java:231]: I think this test is never run with a container. incontainer is false both in https://github.com/Karm/mandrel-integration-tests/pull/196/files#diff-124a2a0b16e9e33fa29c49686e8d5ace8c621908d5db45ff8ed57f827c884555R264 and https://github.com/Karm/mandrel-integration-tests/pull/196/files#diff-124a2a0b16e9e33fa29c49686e8d5ace8c621908d5db45ff8ed57f827c884555R285.

cc @Karm

File: testsuite/src/it/java/org/graalvm/tests/integration/DebugSymbolsTest.java [hunk 3]
```diff
@@ -126,18 +131,23 @@ public void debugSymbolsSmokeGDB(TestInfo testInfo) throws IOException, Interrup
             // Build
             processLog = Path.of(appDir.getAbsolutePath(), "logs", "build-and-run.log").toFile();
 
-            Map<String, String> switches;
-            Version version = UsedVersion.getVersion(app.runtimeContainer != ContainerNames.NONE);
+            Map<String, String> switches = new HashMap<>();
+            Version version = UsedVersion.getVersion(false);
```
Comment (zakkak) [inline testsuite/src/it/java/org/graalvm/tests/integration/DebugSymbolsTest.java:135]: This one also never runs with the builder image (i.e. in a container).

File: testsuite/src/it/java/org/graalvm/tests/integration/DebugSymbolsTest.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (jerboaa) [reply testsuite/src/it/java/org/graalvm/tests/integration/DebugSymbolsTest.java:231]: Fair enough. It just seems an anti-pattern to spread around the code base that's all.

File: testsuite/src/it/java/org/graalvm/tests/integration/utils/BuildAndRunCmds.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (jerboaa) [reply testsuite/src/it/java/org/graalvm/tests/integration/utils/BuildAndRunCmds.java:210]: We can revisit this at a later point.

**Review by jerboaa — APPROVED** [review]
Lets ship this so as to get cleaner CI runs.

**Pull request merged**